nestedtensor/csrc/python_args.h

```diff
@@ -1,44 +1,50 @@
 #include <torch/extension.h>
 
+template <typename T>
 struct IAR {
-  std::vector<int64_t> val;
-  bool is_int;
+  std::vector<T> val;
+  bool not_list;
   template <int64_t repeat>
-  at::IntArrayRef extract() {
-    if (is_int && val.size() < repeat) {
-      for (int64_t i = val.size() ; i < repeat; i++) {
+  at::ArrayRef<T> extract() {
+    if (not_list && val.size() < repeat) {
+      for (T i = val.size(); i < repeat; i++) {
         val.push_back(val[0]);
       }
     }
-    return at::IntArrayRef(val);
+    return at::ArrayRef<T>(val);
   }
 };
 
 namespace pybind11 {
 namespace detail {
-template <>
-struct type_caster<IAR> {
+template <typename T>
+struct type_caster<IAR<T>> {
  public:
   /**
    * This macro establishes the name 'inty' in
    * function signatures and declares a local variable
    * 'value' of type inty
    */
-  PYBIND11_TYPE_CASTER(IAR, _("int[]"));
+  PYBIND11_TYPE_CASTER(IAR<T>, _("int[]"));
 
   /**
    * Conversion part 1 (Python->C++).
    */
   bool load(handle obj, bool) {
     /* Extract PyObject from handle */
-    if (py::isinstance<py::int_>(obj) || py::isinstance<py::float_>(obj)) {
+    if (py::isinstance<py::int_>(obj)) {
```

Comment:
Since you're template on a type you should make this condition on the type. If it's an integral type you should cast it to that.